subsys/net/lib/azure_iot_hub/src/azure_iot_hub.c

```diff
@@ -1110,376 +637,462 @@ static void fota_evt_handler(struct azure_fota_event *fota_evt)
+				return -ETIMEDOUT;
+			}
 
-#if IS_ENABLED(CONFIG_AZURE_IOT_HUB_DPS)
-	struct dps_config cfg = {
-		.mqtt_client = &client,
-		.handler = dps_handler,
-		.reg_id = conn_config.device_id,
-		.reg_id_len = conn_config.device_id_len,
-	};
+			break;
+		case -EALREADY:
+			LOG_DBG("Already assigned to an IoT hub, skipping DPS");
+			break;
+		default:
+			LOG_ERR("azure_iot_hub_dps_start failed, error: %d", err);
+			return err;
+		}
 
-	err = dps_init(&cfg);
-	if (err) {
-		LOG_ERR("Failed to initialize DPS, error: %d", err);
-		return err;
+		err = azure_iot_hub_dps_hostname_get(&conn_params.hostname);
+		if (err) {
+			LOG_ERR("Failed to get the stored hostname from DPS, error: %d", err);
+			return -EFAULT;
+		}
+
+		LOG_DBG("Using the assigned hub (size: %d): %s",
+			conn_params.hostname.size, conn_params.hostname.ptr);
+
+		err = azure_iot_hub_dps_device_id_get(&conn_params.device_id);
+		if (err) {
+			LOG_ERR("Failed to get the stored device ID from DPS, error: %d", err);
+			return -EFAULT;
+		}
+
+		LOG_DBG("Using the assigned device ID: %.*s",
+			conn_params.device_id.size,
+			conn_params.device_id.ptr);
 	}
 
-	LOG_DBG("DPS initialized");
 #endif /* IS_ENABLED(CONFIG_AZURE_IOT_HUB_DPS) */
 
-#if IS_ENABLED(CONFIG_AZURE_FOTA)
-	err = azure_fota_init(fota_evt_handler);
+	err = mqtt_helper_init(&cfg);
 	if (err) {
-		LOG_ERR("Failed to initialize Azure FOTA, error: %d", err);
-		return err;
+		LOG_ERR("mqtt_helper_init failed, error: %d", err);
+		return -EFAULT;
 	}
 
-	LOG_DBG("Azure FOTA initialized");
-#endif
+	hostname_span = az_span_create(conn_params.hostname.ptr, conn_params.hostname.size);
+	device_id_span = az_span_create(conn_params.device_id.ptr, conn_params.device_id.size);
 
-	evt_handler = event_handler;
-	is_initialized = true;
-	(void)err;
+	/* Initialize Azure SDK client instance. */
+	err = az_iot_hub_client_init(
+		&iot_hub_client,
+		hostname_span,
+		device_id_span,
+		NULL);
+	if (az_result_failed(err)) {
+		LOG_ERR("Failed to initialize IoT Hub mqtt_client, result code: %d", err);
+		iot_hub_state_set(STATE_DISCONNECTED);
+
+		return -EFAULT;
+	}
+
+	err = az_iot_hub_client_get_user_name(&iot_hub_client,
+					      user_name_buf,
+					      sizeof(user_name_buf),
+					      &user_name_len);
+	if (az_result_failed(err)) {
+		LOG_ERR("Failed to get user name, az error: 0x%08x", err);
+		return -EFAULT;
+	}
+
+	conn_params.user_name.size = user_name_len;
 
-	connection_state_set(STATE_INIT);
+	LOG_DBG("User name: %.*s", conn_params.user_name.size, conn_params.user_name.ptr);
+	LOG_DBG("User name buffer size is %d, actual user name size is: %d",
+		sizeof(user_name_buf), user_name_len);
+
+	err = mqtt_helper_connect(&conn_params);
+	if (err) {
+		LOG_ERR("mqtt_helper_connect failed, error: %d", err);
+		return err;
+	}
 
 	return 0;
 }
 
-int azure_iot_hub_method_respond(struct azure_iot_hub_result *result)
+int azure_iot_hub_send(const struct azure_iot_hub_msg *const msg)
```

Comment:
This function is also over 100 lines long. Maybe it's possible to extract the body of each switch-case to make it easier to read?